File: apps/api/app/schemas/billing.py
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel


class SetupPaymentResponse(BaseModel):
    client_secret: str


class PaymentMethodSummary(BaseModel):
    id: str
    brand: str | None = None
    last4: str | None = None
    exp_month: int | None = None
    exp_year: int | None = None


class SellerOnboardingResponse(BaseModel):
    onboarding_url: str


class BillingInvoiceSummary(BaseModel):
    id: str
    amount_due: Decimal
    amount_paid: Decimal
    status: str | None = None
    created_at: datetime | None = None
    hosted_invoice_url: str | None = None
    invoice_pdf: str | None = None


class SellerPayoutHistoryItem(BaseModel):
    id: str
    amount: Decimal
    currency: str
    created_at: datetime | None = None
    status: str | None = None


class RevenueByToolItem(BaseModel):
    tool_id: str
    tool_name: str
    revenue: Decimal
    platform_fee: Decimal
    seller_payout: Decimal


class SellerBalanceResponse(BaseModel):
    current_balance: Decimal
    pending_payouts: Decimal
    payout_history: list[SellerPayoutHistoryItem]
    revenue_by_tool: list[RevenueByToolItem]

File: apps/web/src/types/billing.ts
export interface SetupPaymentResponse {
  client_secret: string;
}

export interface PaymentMethodSummary {
  id: string;
  brand: string | null;
  last4: string | null;
  exp_month: number | null;
  exp_year: number | null;
}

export interface ToolPurchaseResponse {
  id: string;
  tool_id: string;
  buyer_id: string;
  seller_id: string;
  status: string;
  purchase_price: string;
  purchase_type: string;
  created_at: string;
  checkout_url: string | null;
}

export interface BillingInvoiceSummary {
  id: string;
  amount_due: string;
  amount_paid: string;
  status: string | null;
  created_at: string | null;
  hosted_invoice_url: string | null;
  invoice_pdf: string | null;
}

export interface SellerPayoutHistoryItem {
  id: string;
  amount: string;
  currency: string;
  created_at: string | null;
  status: string | null;
}

export interface RevenueByToolItem {
  tool_id: string;
  tool_name: string;
  revenue: string;
  platform_fee: string;
  seller_payout: string;
}

export interface SellerBalanceResponse {
  current_balance: string;
  pending_payouts: string;
  payout_history: SellerPayoutHistoryItem[];
  revenue_by_tool: RevenueByToolItem[];
}
